infra/bridge/src/controllers/telegram.js

```diff
@@ -3,98 +3,48 @@
-
-const { telegramGenerateCode, telegramVerify } = require('../utils/validation')
+const { telegramAttestation } = require('../utils/validation')
 
 const { redisClient, getAsync } = require('../utils/redis')
 
-router.get('/generate-code', telegramGenerateCode, async (req, res) => {
+router.post('/generate-code', telegramAttestation, async (req, res) => {
   const identity = req.query.identity.toLowerCase()
 
-  const { code, seed } = generateTelegramCode(identity)
-
-  redisClient.set(`telegram/attestation/seed/${identity}`, seed, 'EX', 60 * 30)
+  // Store IP to redis
+  redisClient.set(
+    `telegram/attestation/${identity}`,
+    JSON.stringify({
+      ip: req.ip
+    }),
+    'EX',
+    60 * 30
+  )
 
   res.send({
-    code: code
+    code: identity
   })
 })
 
-router.post('/verify', telegramVerify, async (req, res) => {
-  const identity = req.body.identity.toLowerCase()
-  const code = req.body.code
+router.get('/status', telegramAttestation, async (req, res) => {
+  const identity = req.query.identity.toLowerCase()
 
-  const eventKey = `telegram/attestation/event/${Web3.utils.sha3(code)}`
-  const seedKey = `telegram/attestation/seed/${identity}`
+  const key = `telegram/attestation/${identity}/status`
 
-  const storedEvent = await getAsync(eventKey)
-  const seed = await getAsync(seedKey)
+  let statusObj = await getAsync(key)
 
-  if (!storedEvent || !seed) {
-    return res.status(500).send({
-      errors: [`You haven't interacted with the verification bot yet.`]
-    })
-  }
+  statusObj = statusObj ? JSON.parse(statusObj) : null
 
-  const { payload, message } = JSON.parse(storedEvent)
+  const verified = !!(statusObj && statusObj.verified && statusObj.attestation)
 
-  if (!verifyTelegramCode(identity, payload, seed)) {
-    return res.status(500).send({
-      errors: [`Code verification failed`]
-    })
+  // Delete from redis, once verified
+  if (verified) {
+    redisClient.del(key)
   }
 
-  const userProfileData = message.from
-  const profileUrl = userProfileData.username
-    ? `https://t.me/${userProfileData.username}`
-    : null
-
-  const attestationBody = {
-    verificationMethod: {
-      oAuth: true
-    },
-    site: {
-      siteName: 'telegram.com',
-      userId: {
-        raw: String(userProfileData.id)
-      },
-      username: {
-        raw: userProfileData.username
-      },
-      profileUrl: {
-        raw: profileUrl
-      }
-    }
-  }
-
-  try {
-    const attestation = await generateAttestation(
-      AttestationTypes.TELEGRAM,
-      attestationBody,
-      {
-        uniqueId: userProfileData.id,
-        username: userProfileData.username,
-        profileUrl: profileUrl,
-        profileData: userProfileData
-      },
-      req.body.identity,
-      req.ip
-    )
-
-    return res.send(attestation)
-  } catch (error) {
-    logger.error(error)
-    return res.status(500).send({
-      errors: ['Could not create attestation.']
-    })
-  }
+  res.status(200).send({
```

Comment:
might make sense to return 404 if statusObj === null

Reply:
I am not sure about that. The purpose of that endpoint is to check if the user has done any Telegram attestation recently. If the `statusObj` is null, it means that the user hasn't done that. In that case, it would return `{ verified: false }`. 

The DApp uses long-polling to fetch this status. However, we don't have an unique id that identifies a particular job (we use user's address instead of that). And we don't log or set the `statusObj` when an user starts to do attestation either. If we did both of these, 404 would be nice. In this case, I personally feel 200 to be more apt.

But, I'm open to changing this though, if needed.

Reply:
Aha thanks makes sense. Querying for a particular resource and returning 404 is fitting. Not so much in this case. 👍